src/main/java/com/bytezone/dm3270/display/ScreenPacker.java

```diff
@@ -27,24 +25,25 @@ public Command readModifiedFields(byte currentAID, int cursorLocation,
       boolean readModifiedAll, boolean sscpLuData) {
     // pack the AID
     int ptr = 0;
+
     if (!sscpLuData) {
-      buffer[ptr++] = currentAID;               // whatever key was pressed
+      // pack the cursor address
+      buffer[ptr++] = currentAID;
+      BufferAddress ba = new BufferAddress(cursorLocation);
+      ptr = ba.packAddress(buffer, ptr);
     }
 
     // PA keys and the CLR key only return the AID byte
     if (!readModifiedAll) {
       if (currentAID == AIDCommand.AID_PA1 || currentAID == AIDCommand.AID_PA2
           || currentAID == AIDCommand.AID_PA3 || currentAID == AIDCommand.AID_CLEAR) {
+        if (sscpLuData) {
```

Comment:
As per our discussion in fact we should always in this case set buffer with currentAID, so we should remove the conditional if.